Refactor picker to be a cross-platform component

tgolen opened the pull request:
@nickmurray47 @NikkiWines This PR was based on some stuff that I came across in https://github.com/Expensify/Expensify.cash/pull/1767, primarily some missing method docs. As I looked more at the PR, I spotted a few things that should be improved to match our existing patterns better. Please have a look at this and let me know if you have any questions

### Details
- Adds some missing method docs
- Renames a prop to have a less redundant name
- DRY up some logic in the checkbox
- Convert the picker into a cross-platform component

### Fixed Issues
No issue, just some things I spotted in https://github.com/Expensify/Expensify.cash/pull/1767

### Tests / QA Steps
1. Go to the profile page
2. Go to preferences
3. Change the priority mode
4. Go back to settings
5. Go to profile
6. Change preferred prounouns
7. Change timezones
8. Change automatic timezone setting
9. Verify all changes are saved  properly

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
![image](https://user-images.githubusercontent.com/1228807/113749844-572a9980-96c7-11eb-9168-fda519cd1518.png)


#### Mobile Web
![image](https://user-images.githubusercontent.com/1228807/114237236-040f4b80-9940-11eb-894e-92342ff5e898.png)


#### Desktop
![image](https://user-images.githubusercontent.com/1228807/113749875-5f82d480-96c7-11eb-859a-8437c61af57d.png)


#### iOS
![image](https://user-images.githubusercontent.com/1228807/113753107-f3a26b00-96ca-11eb-8ea5-1a3b95d1d4be.png)


#### Android
![image](https://user-images.githubusercontent.com/1228807/113749860-5bef4d80-96c7-11eb-9169-aaf98a2bd148.png)

tgolen committed:
Add method docs

tgolen committed:
Rename checkbox callback prop

tgolen committed:
Finish renaming the callback prop

tgolen committed:
Fix proptype warnings

tgolen committed:
Refactor picker into its own component

tgolen committed:
DRY render logic

tgolen committed:
Remove unused component

tgolen committed:
Add more method docs

nickmurray47 commented on src/components/Checkbox.js:
```diff
@@ -22,22 +22,21 @@ const defaultProps = {
-        </Pressable>
-        {label && (
-            <Pressable onPress={() => onCheckboxClick(!isChecked)}>
+        <Pressable onPress={() => onClick(!isChecked)}>
+            {label ? (
```
are we hiding the checkbox from now on if there's a label present? I think we should show the Checkmark icon regardless, otherwise, it's hard to know that it's a clickable element

tgolen replied:
Ah, good catch. I think I read the original code wrong. I need to revert this.

nickmurray47 commented on src/pages/home/report/ReportActionPropTypes.js:
```diff
@@ -4,17 +4,17 @@ import ReportActionFragmentPropTypes from './ReportActionFragmentPropTypes';
 export default {
     // Name of the action e.g. ADDCOMMENT
-    actionName: PropTypes.string.isRequired,
+    actionName: PropTypes.string,
```
why the switch to making these prop optional?

tgolen replied:
There was a prop warning in the console that `undefined` was being passed. I looked through the code and there is a spot where the report action passed is `{}` (an empty object) so I just corrected the propTypes here.

tgolen committed:
Revert the DRY changes for checkbox

tgolen reviewed (commented):
Updated!

nickmurray47 reviewed (dismissed):
LGTM and test well on Web and iOS

nickmurray47 commented:
looks like there are some merge conflicts too @tgolen

tgolen committed:
Merge branch 'master' into tgolen-improve-profile-quality

tgolen committed:
Rename files to match conventions

tgolen commented:
OK, updated to fix the conflicts. I added a file rename so that we are using the proper conventions and pointed out some more missing method docs and comments in https://github.com/Expensify/Expensify.cash/pull/2169

NikkiWines commented on src/components/Picker/index.android.js:
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
```
I'm not sure I understand why we need three different, largely identical files for the picker.

sketchydroide replied:
is the style the only change? could we have that an input for another compnent perharphs? No real idea, just wandering

tgolen replied:
Yeah, the styles are basically the only difference. I like the idea of just exporting the proper styles from the platform-specific files and DRYing this up more. I'll be working on that today.

NikkiWines commented on src/components/Picker/index.android.js:
```diff
@@ -0,0 +1,35 @@
+        items={items}
+        style={useDisabledStyles ? {
+            ...styles.picker,
+            inputAndroid: [
```
This can be 
```JS
inputAndroid: [styles.picker.inputAndroid, styles.textInput, styles.disabledTextInput],
```

Maftalion commented on src/components/Picker/index.android.js:
```diff
@@ -0,0 +1,35 @@
+        items={items}
+        style={useDisabledStyles ? {
+            ...styles.picker,
+            inputAndroid: [
+                styles.picker.inputAndroid, styles.textInput, styles.disabledTextInput,
+            ],
```
not sure how platform specific styles are handled but would it make more sense to have the ios/android/index files just export the style and the component itself be defined once? would avoid duplication and make it easier to update without having to touch 3 files

NikkiWines reviewed (commented):
Two small comments, looking good otherwise.

tgolen committed:
Merge branch 'master' into tgolen-improve-profile-quality

tgolen committed:
Only have platform specific styles

tgolen committed:
Revert changes to lock file

tgolen commented:
OK, updated and retested on all platforms. I refactored it so the component is only defined once and only the styles are platform specific.

tgolen committed:
Merge branch 'master' into tgolen-improve-profile-quality

NikkiWines reviewed (approved):
This looks good to me, any reason that mWeb is marked as n/a?

tgolen commented:
Just because... laziness 😁 I added testing for mobile web.

sketchydroide reviewed (approved):
LGTM

isagoico commented:
We are blocked to check this QA this PR in both iOS and Android apps because of the scrolling issue in settings>profile https://github.com/Expensify/Expensify.cash/issues/2147

isagoico commented:
QA in other platforms was a pass (mweb, web, desktop)
